### PR TITLE
[skip ci] VIC 1.2 vulnerability scanning integration test plan

### LINKED ISSUE #5690: VIC 1.2 vulnerability scanning integration test
User statement:  As a customer of VIC, I want to be assured through test that the new vulnerability scanning feature of 1.2 is working as expected

Details:
https://github.com/vmware/harbor/issues/2042

Acceptance criteria:
Vulnerability scanning integration test is well documented in a test plan and executed either manually or automated

### PR DESCRIPTION
**Pull request by singhshwetaQE**
Fixes #5690

### CODE REVIEW

File: tests/manual-test-cases/Group19-VIC-Product/19-10-VIC-Vulnerability.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+# Environment:
+This test requires that a vSphere server is running and available
+
+# Test Steps:
+1. Download and install the VIC OVA from here:
+[OVA Releases](https://console.cloud.google.com/storage/browser/vic-product-ova-releases/?project=eminent-nation-87317&authuser=1)
+2. Register Harbor as a registry with Admiral
+3. Populate the harbor instance with images that can be pulled
+4. Do a docker login/pull from the harbor instance
+5. Do a docker push to the harbor instance
```
Comment (andrewtchin) [inline tests/manual-test-cases/Group19-VIC-Product/19-10-VIC-Vulnerability.md]: Should this be combined with 3?

File: tests/manual-test-cases/Group19-VIC-Product/19-10-VIC-Vulnerability.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+6. Login to Admiral UI, Go to ‘Administration->Projects->“Repositories” tab’ locate the specified repository
```
Comment (andrewtchin) [inline tests/manual-test-cases/Group19-VIC-Product/19-10-VIC-Vulnerability.md]: Are there supposed to be quotes on Repositories?

File: tests/manual-test-cases/Group19-VIC-Product/19-10-VIC-Vulnerability.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+7. Click on the arrow to expand the embedded tag list of the specified repository and then click on the vertical ellipsis to open the popup menu.
+8. Click on 'Scan' to start the vulnerability analysis process
+9. Verify that if error occurred, user can click on the 'View Log'
+10. Verify different scanning status "Not Scanned" "Queued" "Scanning" "Error" "Complete" - status are indicated in the 'Vulnerability' column
```
Comment (andrewtchin) [inline tests/manual-test-cases/Group19-VIC-Product/19-10-VIC-Vulnerability.md]: Commas needed for the list

File: tests/manual-test-cases/Group19-VIC-Product/19-10-VIC-Vulnerability.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+11. Verify result once the scan has been completed - Red: High , Orange: Medium, Yellow: Low, Grey: Unknown and Green: No vulnerabilities
```
Comment (andrewtchin) [inline tests/manual-test-cases/Group19-VIC-Product/19-10-VIC-Vulnerability.md]: Extra space after `High`
Also, should the user push certain images to get the various statuses?

File: tests/manual-test-cases/Group19-VIC-Product/19-10-VIC-Vulnerability.md (hunk 2 above)
Comment (singhshwetaQE) [inline tests/manual-test-cases/Group19-VIC-Product/19-10-VIC-Vulnerability.md]: I will remove quotes

File: tests/manual-test-cases/Group19-VIC-Product/19-10-VIC-Vulnerability.md (hunk 3 above)
Comment (singhshwetaQE) [inline tests/manual-test-cases/Group19-VIC-Product/19-10-VIC-Vulnerability.md]: I will add commas

File: tests/manual-test-cases/Group19-VIC-Product/19-10-VIC-Vulnerability.md (hunk 4 above)
Comment (singhshwetaQE) [inline tests/manual-test-cases/Group19-VIC-Product/19-10-VIC-Vulnerability.md]: @andrewtchin when we push images in vulnerability scan column it shows status if image is new then it should display Not Scanned and after clicking on scan it displays Queued and then depending on scan result- Completed or Error. That is how it should work. 

I will remove extra space

File: tests/manual-test-cases/Group19-VIC-Product/19-10-VIC-Vulnerability.md (hunk 1 above)
Comment (singhshwetaQE) [inline tests/manual-test-cases/Group19-VIC-Product/19-10-VIC-Vulnerability.md]: keeping separate step gives more clarity

File: tests/manual-test-cases/Group19-VIC-Product/19-10-VIC-Vulnerability.md (hunk 4 above)
Comment (andrewtchin) [inline tests/manual-test-cases/Group19-VIC-Product/19-10-VIC-Vulnerability.md]: My question was intended to address if we want to test each of the result types. If we do, then we should specify that an image for each status should be uploaded and scanned.

File: tests/manual-test-cases/Group19-VIC-Product/19-10-VIC-Vulnerability.md (hunk 1 above)
Comment (andrewtchin) [inline tests/manual-test-cases/Group19-VIC-Product/19-10-VIC-Vulnerability.md]: But aren't they the same thing? Populating the Harbor registry means pushing images to the Harbor registry, no?

File: tests/manual-test-cases/Group19-VIC-Product/19-10-VIC-Vulnerability.md (hunk 4 above)
Comment (singhshwetaQE) [inline tests/manual-test-cases/Group19-VIC-Product/19-10-VIC-Vulnerability.md]: Yes we can test that, we can work on same image for all the status except Completed or Error.
I can separate those two steps  

10. Not Scanned->Queued->Scanning->Completed
11.Not Scanned->Queued->Scanning->Error

File: tests/manual-test-cases/Group19-VIC-Product/19-10-VIC-Vulnerability.md (hunk 1 above)
Comment (singhshwetaQE) [inline tests/manual-test-cases/Group19-VIC-Product/19-10-VIC-Vulnerability.md]: Yes I can combine them

File: tests/manual-test-cases/Group19-VIC-Product/19-10-VIC-Vulnerability.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+1. Download and install the VIC OVA from here:
+[OVA Releases](https://console.cloud.google.com/storage/browser/vic-product-ova-releases/?project=eminent-nation-87317&authuser=1)
+2. Register Harbor as a registry with Admiral
+3. Do a docker push to the harbor instance and populate the harbor instance with images that can be pulled
+4. Do a docker login/pull from the harbor instance
+5. Login to Admiral UI, Go to ‘Administration->Projects->Repositories tab’ locate the specified repository
+6. Click on the arrow to expand the embedded tag list of the specified repository and then click on the vertical ellipsis to open the popup menu.
+7. Click on 'Scan' to start the vulnerability analysis process
+8. Verify different scanning status "Not Scanned","Queued","Scanning" and "Complete"- status are indicated in the 'Vulnerability' column
+9. Verify different scanning status "Not Scanned","Queued","Scanning" and "Error" - status are indicated in the 'Vulnerability' column
```
Comment (andrewtchin) [inline tests/manual-test-cases/Group19-VIC-Product/19-10-VIC-Vulnerability.md]: It's not clear what image the status in 8 and 9 should apply to

File: tests/manual-test-cases/Group19-VIC-Product/19-10-VIC-Vulnerability.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+10. Verify that if error occurred, user can click on the 'View Log'
+11. Verify result once the scan has been completed - Red: High, Orange: Medium, Yellow: Low, Grey: Unknown and Green: No vulnerabilities
```
Comment (andrewtchin) [inline tests/manual-test-cases/Group19-VIC-Product/19-10-VIC-Vulnerability.md]: It's not clear how to check that each of these statuses is present. Should there be a different image for each one or is there a procedure to trigger each value?

Comment (singhshwetaQE) [inline tests/manual-test-cases/Group19-VIC-Product/19-10-VIC-Vulnerability.md]: @andrewtchin Verify the result once the scan has been completed, look for color bar.

File: tests/manual-test-cases/Group19-VIC-Product/19-10-VIC-Vulnerability.md (hunk 2 above)
Comment (steven-zou) [inline tests/manual-test-cases/Group19-VIC-Product/19-10-VIC-Vulnerability.md]: Go to ‘Administration->Projects->“Repositories” tab’ 
===>
Go to ‘Administration->Projects->[Your Project] -> Repositories’

**Pull request merged**